libs/mbedtls/mbedtls_stubs.c

```diff
@@ -152,6 +153,79 @@ CAMLprim value ml_mbedtls_x509_crt_parse_path(value chain, value path) {
 	CAMLreturn(Val_int(mbedtls_x509_crt_parse_path(X509Crt_val(chain), String_val(path))));
 }
 
+// Certificate Haxe API
+
+value caml_string_of_asn1_buf(mbedtls_asn1_buf* dat) {
+	CAMLparam0();
+	CAMLlocal1(s);
+	s = caml_alloc_string(dat->len);
+	memcpy(String_val(s), dat->p, dat->len);
+	CAMLreturn(s);
+}
+
+CAMLprim value hx_cert_get_subject(value chain, value objname) {
+	CAMLparam2(chain, objname);
+	mbedtls_x509_name *obj;
+	mbedtls_x509_crt* cert = X509Crt_val(chain);
+	int r;
```

Comment:
Why forward declare this?

Reply:
I remember getting some warnings about having variable declarations that aren't at the beginning of the function, but maybe that was in a different context.